File: src/wasteApp_server/src/main/java/com/fantastiSquad/wasteApp/models/entities/City.java
package com.fantastiSquad.wasteApp.models.entities;

import static javax.persistence.GenerationType.IDENTITY;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

@Entity
@Table(name="city")
public class City {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String name;
  private String citycode;

  @OneToMany(fetch = FetchType.EAGER, cascade = {CascadeType.PERSIST, CascadeType.REMOVE})
  @Fetch(FetchMode.SELECT)
  private Set<BinTarget> binTarget = new HashSet<>();

  public City(Long id, String name, String citycode,
      Set<BinTarget> binTarget) {
    this.id = id;
    this.name = name;
    this.citycode = citycode;
    this.binTarget = binTarget;
  }

  public City(String name, String citycode,
      Set<BinTarget> binTarget) {
    this.name = name;
    this.citycode = citycode;
    this.binTarget = binTarget;
  }

  public City() {
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getname() {
    return name;
  }

  public void setname(String name) {
    this.name = name;
  }

  public String getcitycode() {
    return citycode;
  }

  public void setcitycode(String citycode) {
    this.citycode = citycode;
  }

  public Set<BinTarget> getBinTarget() {
    return binTarget;
  }

  public void setBinTarget(Set<BinTarget> binTarget) {
    this.binTarget = binTarget;
  }

  @Override
  public String toString() {
    return "City{" +
        "id=" + id +
        ", name='" + name + '\'' +
        ", citycode='" + citycode + '\'' +
        ", binTarget=" + binTarget +
        '}';
  }
}

File: src/wasteApp_client/wasteApp_mob_front/src/app/shared/models/city.ts
import { BinTarget } from './bin-target';

export class City {
    private id: number; 
    private name: String;
    private citycode: String;
    private binTarget: BinTarget [];
    
    constructor(name: String, citycode: String, binTarget: BinTarget []){
        
        this.name = name;
        this.citycode = citycode;
        this.binTarget = binTarget;
    }
    /*Getters and Setters */
    
    public get getid(): number {
        return this.id;
    }
    public set setid(value: number) {
        this.id = value;
    }
    
    /**
     * Getter name
     * @return {String}
     */
	public get getname(): String {
		return this.name;
	}

    /**
     * Getter cityCode
     * @return {String}
     */
	public get getcitycode(): String {
		return this.citycode;
	}

    /**
     * Getter binTarget
     * @return {Object []}
     */
	public get getbinTarget(): BinTarget [] {
		return this.binTarget;
	}

    /**
     * Setter name
     * @param {String} value
     */
	public set setname(value: String) {
		this.name = value;
	}

    /**
     * Setter cityCode
     * @param {String} value
     */
	public set setcitycode(value: String) {
		this.citycode = value;
	}

    /**
     * Setter binTarget
     * @param {Object []} value
     */
	public set setbinTarget(value: BinTarget []) {
		this.binTarget = value;
	}   

}

